misaligned.c

```diff
@@ -1,21 +1,61 @@
 #include <stdio.h>
 #include <assert.h>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+enum MajorColor {WHITE, RED, BLACK, YELLOW, VIOLET};
+enum MinorColor {BLUE, ORANGE, GREEN, BROWN, SLATE};
+
+const char* majorColorNames[] = {"White", "Red", "Black", "Yellow", "Violet"};
+const char* minorColorNames[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+
+int numberOfMajorColors =
+    sizeof(majorColorNames) / sizeof(majorColorNames[0]);
+
+int numberOfMinorColors =
+    sizeof(minorColorNames) / sizeof(minorColorNames[0]);
+
+typedef struct {
+    enum MajorColor majorColorValue;
+    enum MinorColor minorColorValue;
+} ColorPair;
+
+
+
+int printColorMap() {    
     int i = 0, j = 0;
     for(i = 0; i < 5; i++) {
         for(j = 0; j < 5; j++) {
-            printf("%d | %s | %s\n", i * 5 + j, majorColor[i], minorColor[i]);
+            printf("%d | %s | %s\n", ((i*5)+j+1), majorColorNames[i], minorColorNames[j]);

```

Comment:
The formula `(i*5)+j+1` is not testable, because it is combined with a non-pure (printf) function.
Separate it into a pure function and write tests for it.